app/models/course/material/folder.rb

```diff
@@ -8,6 +8,7 @@ class Course::Material::Folder < ActiveRecord::Base
   has_many :materials, inverse_of: :folder, dependent: :destroy, foreign_key: :folder_id,
                        class_name: Course::Material.name
   belongs_to :course, inverse_of: :material_folders
+  belongs_to :owner, polymorphic: true, inverse_of: :folder
```

Comment:
@lowjoel I think actable is a good solution too, thus I we don't need to manually create a folder before uploading. But I guess there are some issues ? Like when and how we set the folder name( or other required fields). Let me think again and get back to you !